db/crud.go

```diff
@@ -1129,20 +1116,28 @@ func (db *DatabaseCollectionWithUser) PutExistingCurrentVersion(ctx context.Cont
 			}
 		}
 
+		// set up revTreeID for backward compatibility
+		previousRevTreeID := doc.CurrentRev
+		prevGeneration, _ := ParseRevID(ctx, previousRevTreeID)
+		newGeneration := prevGeneration + 1
+
 		// Conflict check here
 		// if doc has no HLV defined this is a new doc we haven't seen before, skip conflict check
 		if doc.HLV == nil {
-			doc.HLV = &HybridLogicalVector{}
-			addNewerVersionsErr := doc.HLV.AddNewerVersions(docHLV)
+			newHLV := NewHybridLogicalVector()
+			doc.HLV = &newHLV
+			addNewerVersionsErr := doc.HLV.AddNewerVersions(newDocHLV)
 			if addNewerVersionsErr != nil {
 				return nil, nil, false, nil, addNewerVersionsErr
 			}
 		} else {
-			incomingDecodedHLV := docHLV.ToDecodedHybridLogicalVector()
-			localDecodedHLV := doc.HLV.ToDecodedHybridLogicalVector()
-			if !incomingDecodedHLV.IsInConflict(localDecodedHLV) {
+			if (*doc.HLV).isDominating(newDocHLV) {
```

Comment:
nit: method call will auto-deref the pointer

```suggestion
			if doc.HLV.isDominating(newDocHLV) {
```